tmail-backend/webadmin/webadmin-mailbox/src/main/java/com/linagora/tmail/webadmin/cleanup/CleanupTrashService.java

```diff
@@ -1,12 +1,97 @@
+import java.util.Date;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.apache.james.core.Username;
+import org.apache.james.mailbox.DefaultMailboxes;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.SessionProvider;
+import org.apache.james.mailbox.model.FetchGroup;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.MessageRange;
+import org.apache.james.mailbox.store.StoreMailboxManager;
 import org.apache.james.task.Task;
+import org.apache.james.user.api.UsersRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import com.linagora.tmail.james.jmap.settings.JmapSettings;
+import com.linagora.tmail.james.jmap.settings.JmapSettingsRepository;
+
+import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 public class CleanupTrashService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CleanupTrashService.class);
+    private static final String WEEKLY_PERIOD = "weekly";
+    private static final String MONTHLY_PERIOD = "monthly";
+    private static final String TRASH_CLEANUP_ENABLED_SETTING = "trash.cleanup.enabled";
+    private static final String TRASH_CLEANUP_PERIOD_SETTING = "trash.cleanup.period";
+    private final UsersRepository usersRepository;
+    private final JmapSettingsRepository jmapSettingsRepository;
+    private final StoreMailboxManager storeMailboxManager;
+    private final SessionProvider sessionProvider;
+
+    @Inject
+    public CleanupTrashService(UsersRepository usersRepository, JmapSettingsRepository jmapSettingsRepository, StoreMailboxManager storeMailboxManager, SessionProvider sessionProvider) {
+        this.usersRepository = usersRepository;
+        this.jmapSettingsRepository = jmapSettingsRepository;
+        this.storeMailboxManager = storeMailboxManager;
+        this.sessionProvider = sessionProvider;
+    }
 
     public Mono<Task.Result> cleanupTrash() {
-        return Mono.just(Task.Result.COMPLETED);
+        return Flux.from(usersRepository.listReactive())
+            .flatMap(this::checkSettingAndCleanupTrashForSingleUser)
+            .reduce(Task.Result.COMPLETED, Task::combine);
+    }
+
+    private Mono<Task.Result> checkSettingAndCleanupTrashForSingleUser(Username username) {
+        return Mono.from(jmapSettingsRepository.get(username))
+            .flatMap(jmapSettings -> getJMAPSettingsValue(jmapSettings, TRASH_CLEANUP_ENABLED_SETTING)
+                .map(trashCleanupEnabled -> {
+                    if ("true".equals(trashCleanupEnabled.value())) {
+                        return getJMAPSettingsValue(jmapSettings, TRASH_CLEANUP_PERIOD_SETTING)
+                            .map(trashCleanupPeriod -> cleanupTrashForSingleUser(username, trashCleanupPeriod.value()))
+                            .orElseThrow(() -> new RuntimeException("The setting value of trash.cleanup.period does not exist"));
```

Comment:
You should rather return an error Mono this is more idiomatic of reactive programming / functional programming.